spark/v3.4/spark/src/test/java/org/apache/iceberg/spark/actions/TestBaseSparkAction.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg.spark.actions;
+
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+import org.apache.iceberg.spark.JobGroupInfo;
+import org.apache.iceberg.spark.SparkTestBase;
+import org.apache.spark.scheduler.SparkListener;
+import org.apache.spark.scheduler.SparkListenerJobStart;
+import org.apache.spark.sql.SparkSession;
+import org.junit.Assert;
+import org.junit.Test;
+
+public class TestBaseSparkAction extends SparkTestBase {
+
+  @Test
+  public void testCancelSparkActionJob() throws InterruptedException {
+    String groupId = "test-group-001";
+    AtomicReference<Boolean> result = new AtomicReference<>();
+    CountDownLatch latch = new CountDownLatch(1);
+    spark
+        .sparkContext()
+        .addSparkListener(
+            new SparkListener() {
+              @Override
+              public void onJobStart(SparkListenerJobStart jobStart) {
+                latch.countDown();
+              }
+            });
+    Thread thread =
+        new Thread(
+            () -> {
+              sparkContext.setJobGroup(groupId, "test-cancel-spark-action-job", true);
+              String expectedErrMsg = "cancelled job group " + groupId;
+              SleepSparkAction action = new SleepSparkAction(spark, 10000L, expectedErrMsg);
```

Comment:
10s sleep is too long for a single ut, could you change to use await or similar to improve it?

Reply:
I see, will interrupt shortly

Reply:
yes, the job will be canceled after receiving SparkListenerJobStart event